gridstatus/pjm.py

```diff
@@ -1693,3 +1693,134 @@ def _parse_transmission_limits(self, df: pd.DataFrame):
         ]
 
         return df.sort_values("Interval Start").reset_index(drop=True)
+
+    @support_date_range(frequency=None)
+    def get_solar_generation_by_area(
+        self,
+        date: str | pd.Timestamp,
+        end: Optional[str | pd.Timestamp] = None,
+        verbose: Optional[bool] = False,
+    ):
+        """
+        Retrieves the current solar generation information from:
+        https://dataminer2.pjm.com/feed/solar_gen/definition
+        Data is published daily around 7am market time.
+
+        Arguments:
+            date (str or pandas.Timestamp): Start datetime for data
+            end: (str or pandas.Timestamp, optional): End datetime for data.
+                Defaults to one day past `date` if not specified.
+            verbose (bool, optional): print verbose output. Defaults to False.
+
+        Returns:
+            pandas.DataFrame: A DataFrame with solar generation information.
+        """
+        if date == "latest":
+            date = "today"
+
+        df = self._get_pjm_json(
+            "solar_gen",
+            start=date,
+            params={
+                "fields": "datetime_beginning_ept,datetime_beginning_utc,"
+                "area,solar_generation_mw",
+            },
+            end=end,
+            filter_timestamp_name="datetime_beginning",
+            interval_duration_min=60,
+            verbose=verbose,
+        )
+
+        return self._parse_solar_generation_by_area(df)
+
+    def _parse_solar_generation_by_area(self, df: pd.DataFrame):
+        df = df.pivot_table(
+            index=["Time", "Interval Start", "Interval End"],
+            columns="area",
+            values="solar_generation_mw",
+            aggfunc="first",
+        ).reset_index()
+
+        df = df[
+            [
+                "Interval Start",
+                "Interval End",
+                "MIDATL",
+                "OTHER",
+                "RFC",
+                "RTO",
+                "SOUTH",
+                "WEST",
```

Comment:
Does solar have the same issue with some areas not present in the historical data as wind?

Reply:
No, it does not. It also does not go back as far, only to 2019 instead of 2011.